design-reference/architecture/VersionAndGo/readme.md

```diff
@@ -0,0 +1,113 @@
+
+ ```
+ module "az-isolated-service-single-region" {
+   source = "git://dev.azure.com/project/repo?ver=1.0.0//infra/template/az-isolated-service-single-region/"
+   resource_group_location = "eastus"
+   name = "az-simple"
+   deployment_targets = [{
+     app_name = "cobalt-backend-api",
+     image_name = "msftcse/az-service-single-region",
+     image_release_tag_prefix = "release"
+   }]
+   acr_build_git_source_url = "https://github.com/erikschlegel/echo-server.git"
+}
+ ```
+
+ - `Application.yaml` - is the artifact that points to the azure-pipeline.yml file in the eng repo that contains the build/test/deploy pipeline.
+
+This folder/files should be checked into the root of the app repo. The Eng team would then configure a new pipeline in the app teams Azure Devops project to point to these files and deploy the requested infrastructure.
+
+> Note: We should strive to continue to use a single set of pipeline files for all pipelines referenced in this document.
+
+## Version and Go Eng Pipeline Description
+
+This pipeline will produce an artifact that is stored in a location that would be accessible by both the engineering team and the app team. It should follow or use the same pipeline as the development team and app team pipelines. This code that publishes a new version could be controlled by a feature flag as global variable in the eng pipeline. This could also use a SemVer global variable and only create a new artifact when this has been incremented. Would could also leave this as a "manual" process. Manual only means that the pipeline doesn't create the artifact. We could use a script or the likes to do this.
+
+## Version and Go App Pipeline Description
+
+The pipeline "bootstrap" file: `Application.yaml` should perform the following...
+
+1. Request to copy the pipeline files down to the build agent from the engineering repo.
+2. It should use the local `Application.tf` file to "bootstrap" the template and download all needed template/module files.
+3. Deploy the template it that is referenced by the `Application.tf` file using the pipeline referenced by the `Application.yml` file.
+
+# Risks and Unknowns
+
+1. Local relative path used in the template to source the modules may fail if we do not copy the module files to the build agent.
+2. We need to add specific functionality to the engineering pipeline to publish versioned artifacts.
+3. We need to be able to copy the same version of pipeline files as the template.
+4. We need to configure the pipeline to use `Application.tf` not template `main.tf ` to deploy from the app repo.
+5. We need to understand how to version using Azure Repos. 
+6. We need to verify we can link to the version "tag" of the Azure Repo from the `source` property in the `Application.tf` file.
+
+# Out-of-Scope
+
+We are currently not including the following items.
+
+  1. Containerization of the commands executed by the pipeline.
+
+# Sequence Diagram
```

Comment:
Need to add information on how to update engineering repo and version. How does the app team update?